src/consensus/aft/raft.h

```diff
@@ -2393,11 +2529,12 @@ namespace aft
 
     void become_candidate()
     {
-      if (replica_state == kv::ReplicaState::Retired)
+      if (replica_state == kv::ReplicaState::Retired || retiring)
```

Comment:
I'm wary of having a few bools in addition to the replica state machine. It might be the right approach, but we should confirm that.